serverless/messaging/reference-content/migration.mdx

```diff
@@ -0,0 +1,173 @@
+
+Billing will be based on:
+
+- For **NATS**: **Stream volume** (the number of messages sent * the message size) and **stream persistence** (the total amount of data stored * duration)
+- For **SQS**: **Queue volume** (the number of messages sent * the message size)
+
+Full details of price points and how the billing model works will be available at the time of General Availability.
+
+<Message type="important">
+The precise start of the billing period for NATS and SQS is defined as the time of General Availability, **not** as the time of the migration to the new `v1beta1` "no-namespace" version of the product. This means that messages sent using both the old and new versions will be billed from GA onwards. Check out our [Slack channel](https://slack.scaleway.com/) #messaging-queuing-beta to stay up to date. The announcement of General Availability will also be visible in the Scaleway console.
+</Message>
+
+Note that **SNS** will not yet move into General Availability, but will instead stay in Public Beta while we continue to work on testing and improving this product. This means that you will not yet be billed for using Messaging and Queuing with the SNS protocol, even though the `v1beta1` version will also be used for SNS.
+
+## Migration: timescales and action to take
+
+Below we explain for each protocol how the migration from the old to the new Messaging and Queuing will work, and what action you will need to take.
+
+### NATS
+
+Scaleway will automatically migrate each of your NATS namespaces and its credentials from the old version of Messaging and Queuing to the new one. In the new version, each namespace will effectively be a NATS account (each with its corresponding credentials). 
+
+The migration should be seamless: if you log into the Scaleway console before migration/GA you will see the "old" version of Messaging and Queuing with your existing resources showing as namespaces. After migration/GA, you will see the "new" version with your newly migrated resources now showing as accounts.
+
+In terms of the API, the [current `v1alpha1`](https://www.scaleway.com/en/developers/api/messaging-and-queuing/) version will continue to be available 30 days after migration/General Availability, co-existing with the new `v1beta1` version. The documentation for both will be accessible via the [Scaleway Developers website](https://www.scaleway.com/en/developers/api) via a dropdown at the top of the Messaging and Queuing API page. 
+
+Internally, NATS namespaces (and their credentials) and NATS accounts (and their credentials) are the same resources, visible through both APIs. During the migration period, both APIs' endpoints are essentially accessing the same resources. Do not attempt to delete your NATS resources from the `v1alpha1` API as this will also delete them from the new `v1beta1` API and console. You will not be billed "twice" for resources in the old and new API, because these are essentially the same resources.
```

Comment:
```suggestion
Internally, NATS namespaces (and their credentials) and NATS accounts (and their credentials) are the same resources, visible through both APIs. During the migration period, both APIs' endpoints are essentially accessing the same resources. Do not attempt to delete your NATS resources from the `v1alpha1` API, as this will also delete them from the new `v1beta1` API and console. You will not be billed "twice" for resources in the old and new API because these are essentially the same resources.
```